lib/api/core/entrypoints/embedded/protocols/socketio.js

```diff
@@ -166,36 +165,40 @@ class SocketIoProtocol extends Protocol {
   }
 
   notify(data) {
-    debug('notify: %a', data);
+    const socket = this.sockets.get(data.connectionId);
 
-    if (this.sockets[data.connectionId]) {
+    if (socket) {
+      debug('notify: %a', data);
       data.channels.forEach(channel => {
```

Comment:
Using `for (let i..)` is better for performance and IMHO this can be considered as a critical section of code

Reply:
I took a look at the whole code and modernized this snippet and other parts as well (see my boyscout commit for a list of changes)